src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1560,26 +1722,41 @@ subroutine ocn_vert_transport_velocity_top(meshPool, verticalMeshPool, scratchPo
       ! thickness-weighted divergence and barotropic divergence
       !
       ! See Ringler et al. (2010) jcp paper, eqn 19, 21, and fig. 3.
+#ifdef MPAS_OPENACC
+      !$acc enter data create(div_hu, projectedSSH, ALE_Thickness)
+
+! DEBUG, MDT :: need to add OpenACC
+!      !$acc parallel loop &
+!      !$acc          present(div_hu, invAreaCell, nEdgesOnCell, edgesOnCell, maxLevelEdgeTop, &
+!      !$acc                  layerThicknessEdge, normalVelocity, dvEdge, edgeSignOnCell, &
+!      !$acc                  oldSSH, projectedSSH) &
+!      !$acc          private(div_hu_btr, invAreaCell1, i, iEdge, flux, k)
```

Comment:
@philipwjones If I uncomment this directive (and the directives around line 1754), then I get a segfault when running on GPUs.  The error looks to be the result of an incorrect calculation of the values in this subroutine.  I've deleted and re-generated this directive numerous times and always end up with the same error.  Could you check the directive to see if you can notice a problem?

Reply:
looking into this now - I have had trouble occasionally with PGI getting confused about these reductions so we'll see if that's the issue and come up with a workaround

Reply:
Update - the loop itself is computing correctly. There is a transfer bug later in the routine. Now that you've rebased, I'll update to the latest and then continue the review and let you know

Reply:
@philipwjones Thanks for looking at this.  I had gone through the subroutines loop by loop, verifying that the overall output from the test was within roundoff of the baseline.  Are you saying that there is an incorrect data transfer elsewhere that is only showing up as a bug when I add OpenACC to this loop?  If so, any suggestions on how to debug such an issue?

Reply:
Actually it isn't really a bug. When I uncommented that loop, I forgot to deal with a later data update of div_hu that's only needed if this loop in done on cpu. When I took care of that, this routine runs fine for me in a QU240 test on Summit.  However, there must still be some issues elsewhere because this same QU240 test is leading to some very wrong answers with OPENACC on. Still trying to track that down. Sorry for the confusion.